File: regexlib/2021-5-15/javascript_test_file/regexlib_8593.js
/* 8593
 * ^((\.)?([a-zA-Z0-9_-]?)(\.)?([a-zA-Z0-9_-]?)(\.)?)+$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"-"*16+"◎@! _1_EOA(i or ii)"
 */
var REGEX = /^((\.)?([a-zA-Z0-9_-]?)(\.)?([a-zA-Z0-9_-]?)(\.)?)+$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '-'.repeat(i*1) + '◎@! _1_EOA(i or ii)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re2_test_file/regexlib_8593.py
# 8593
# ^((\.)?([a-zA-Z0-9_-]?)(\.)?([a-zA-Z0-9_-]?)(\.)?)+$
# EXPONENT
# nums:5
# EXPONENT AttackString:""+"-"*16+"◎@! _1_EOA(i or ii)"

import re2 as re
from time import perf_counter

regex = """^((\.)?([a-zA-Z0-9_-]?)(\.)?([a-zA-Z0-9_-]?)(\.)?)+$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "-" * i * 1 + "◎@! _1_EOA(i or ii)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")